packages/common/src/featureServiceHelpers.ts

```diff
@@ -1539,6 +1541,58 @@ export function validateSpatialReferenceAndExtent(
   }
 }
 
+/**
+ * Get the spatial reference from a views source.
+ * Optionally ensure that the source has a valid geometry type prior to using its values.
+ *
+ * @param serviceInfo Basic service information
+ * @param itemTemplate The current template to process
+ * @param templateDictionary Hash mapping Solution source id to id of its clone (and name & URL for feature service)
+ * @param validateGeom When true the source must contain a geometryType for values to be returned
+ * @private
```

Comment:
Please add return doc.
Is it significant to doc that (after passing the has-geometry test if `validateGeom` is true) what is returned is the first view that matches having _both_ a SR and extent?

Reply:
Just checking for both in the [].some to know if we can bail early.